OnRemoveConnection

beddowdm01 opened the pull request:
Needed a way to override OnRemoveConnection from a class that inherits from Node like can be done with OnAddConnection. Small change but makes it so much nicer when I need to add functionality when a connection is removed.

beddowdm01 committed:
Now I can call override OnRemoveConnection

Seneral commented on Node_Editor_Framework/Runtime/Framework/Core/ConnectionPort.cs:
```diff
@@ -248,6 +248,8 @@ public void RemoveConnection (ConnectionPort port, bool silent = false)
 					"Delete Connection");
+					
+				port.body.OnRemoveConnection(this, port);
 			}
```
Can you move this down to the other callback? Only functional difference is the editor check, but is also more consistent.

beddowdm01 replied:
Yeah you're right! Just finished adjusting it, let me know if there's any other changes you'd prefer.

Seneral commented:
Sorry for the late reply. Thanks, no idea why I added OnAddConnection but no remove equivalent. Just one minor change if possible, for consistency and runtime.

DNStar committed:
PR Comment

DNStar committed:
PR Comment

Same as functioanlity as previous commit except removed accidental refactor changes

DNStar committed:
new line

Seneral commented:
Actually, there are two other things, sorry I didn't see it at first.
First, right now the callback will get the parameters switched, since you're calling it on the connection's body.
Second, both port bodies will want the same notification, so we'd need another line:
```
this.body.OnRemoveConnection(this, port);
port.body.OnRemoveConnection(port, this);
```
If in your case the callback should only be received on the one side (e.g. only on output ports), then you can do that by checking the port directions. Other than that there is currently no guaranteed order on which port the connection is removed from.

DNStar committed:
PR Comment - Notifies both port bodies now

beddowdm01 commented:
Yeah, makes sense! Apologies for overlooking that :)

Seneral commented:
No worries, same for me, thanks for the contribution!
